### ISSUE TITLE
client-go should ignore the ResourceVersion of DELETE event when watching

### ISSUE THREAD

**Issue by choury** [kubernetes/kubernetes#63356]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

The ResourceVersion of DELETE event is not the current version, and may be a far early one. If using it to re-watch [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L394), some events will be received repeatedly.

In some case, kube-scheduler's cache can be "updated" to a former status and generate error:
```
 scheduler cache AssumePod failed: pod xxx/xxx state wasn't initial but get assumed
``` 
or 
```
Operation cannot be fulfilled on pods/binding "xxx": pod xxx is already assigned to node "10.10.10.10"
```
**What you expected to happen**:

A line similar
```
newResourceVersion = *resourceVersion
```
should be added after [L409](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L409)

**Comment by choury** [comment]
@kubernetes/sig-api-machinery-bugs @kubernetes/sig-scheduling-misc

**Comment by lavalamp** [comment]
@cheftako @wenjiaswe

**Comment by wenjiaswe** [comment]
I think this issue is already fixed with #58547. Apiserver watch cache filtering is returning the previous object content, but with the event's resource version is set correctly. cc @roycaihw for confirmation. THanks!

**Comment by liggitt** [comment]
yes, this is the same issue as https://github.com/kubernetes/kubernetes/issues/58545 which was fixed in 1.7+

can you verify your apiserver version?

**Comment by choury** [comment]
The version we using is 1.8.5, I found the fix in 1.8.8 , we will upgrade to 1.8.12 to check it .
